### PR TITLE
Responses key supports both string, int, and HTTPStatus

### PR DESCRIPTION
**Pull request by luolingchun**
Checklist:

- [x] Run `pytest tests` and no failed.
- [x] Run `flake8 flask_openapi3 tests examples` and no failed.
- [x] Run `mypy flask_openapi3` and no failed.
- [x] Run `mkdocs serve` and no failed.

fix: #86

### CODE REVIEW

File: docs/Usage/Response.md [hunk 1]
```diff
@@ -29,6 +29,28 @@ def get_book(path: BookPath, query: BookBody):
...
+*New in v2.5.0*
+
+Now you can use `string`, `int`, and `HTTPStatus` as response's key.
+
+```python hl_lines="5 7"
+from http import HTTPStatus
+
+@api.get("/hello/<string:name>",
+        responses={
+            HTTPStatus.OK: Message,
```
Comment (CostcoFanboy) [inline docs/Usage/Response.md:41]: Odd typehint. You're annotating HTTPStatus.OK with "Message", but I do not see anywhere this Message class.

I'd assume "Message" would be a Union[str, int, HTTPStatus], is the Message class declared somewhere as a type alias?

e.g.
`Message = Union[str, int, HTTPStatus]`

Comment (CostcoFanboy) [reply docs/Usage/Response.md:41]: I just found it here:
examples/response_demo.py

But this means the class is inaccurate in the demo. Therefore this file needs changing too.

File: docs/Usage/Response.md [hunk 2]
```diff
@@ -29,6 +29,28 @@ def get_book(path: BookPath, query: BookBody):
...
+        responses={
+            HTTPStatus.OK: Message, 
+            "201": {"content": {"text/csv": {"schema": {"type": "string"}}}},
+            204: None
+        })
+def hello(path: HelloPath):
+    message = {"message": f"""Hello {path.name}!"""}
+
+    response = make_response(json.dumps(message), HTTPStatus.OK)
+    # response = make_response("sss", HTTPStatus.OK)
```
Comment (CostcoFanboy) [inline docs/Usage/Response.md:49]: Leftover debug?

File: flask_openapi3/_http.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# @Author  : llc
+# @Time    : 2021/6/21 15:54
+from enum import Enum
+from http import HTTPStatus
+
+HTTP_STATUS = {str(status.value): status.phrase for status in HTTPStatus}
```
Comment (CostcoFanboy) [inline flask_openapi3/_http.py:7]: Why do that though 🤔 you can just get rid of this entirely. It's not used anywhere despite this being labelled an internal module.
Not a blocker just curious.

File: flask_openapi3/scaffold.py [hunk 4]
```diff
@@ -48,7 +49,7 @@ def _do_decorator(
             operation_id: Optional[str] = None,
             extra_form: Optional[ExtraRequestBody] = None,
             extra_body: Optional[ExtraRequestBody] = None,
-            responses: Optional[Dict[str, Union[Type[BaseModel], Dict[Any, Any], None]]] = None,
+            responses: Optional[ResponseDict] = None,
```
Comment (CostcoFanboy) [inline flask_openapi3/scaffold.py:52]: Why not use inheritance for GET, PUT, DELETE, etc and only overwrite the http method?
Every time there is a change, you need to change like 5-6 classes. It's inconvenient for you.

File: flask_openapi3/utils.py [hunk 5]
```diff
@@ -283,13 +285,11 @@ def parse_body(
 
 
 def get_responses(
-        responses: Optional[Dict[str, Union[Type[BaseModel], Dict[Any, Any], None]]],
+        responses: ResponseStrKeyDict,
```
Comment (CostcoFanboy) [inline flask_openapi3/utils.py:288]: Great change way easier to read 👍🏻

**Review by CostcoFanboy — CHANGES_REQUESTED** [review]
Few questions.

File: docs/Usage/Response.md (hunk 1 above)
Comment (luolingchun) [reply docs/Usage/Response.md:41]: `HTTPStatus.OK: Message` is not a type of hint. `HTTPStatus.OK` is key and `Message` is value.

And I will improve the documentation here.

Comment (CostcoFanboy) [reply docs/Usage/Response.md:41]: Apologies I was tired and didn't read properly. Please ignore!

File: flask_openapi3/_http.py (hunk 3 above)
Comment (luolingchun) [reply flask_openapi3/_http.py:7]: I just want to be consistent with the built-in modules, `HTTPStatus` is not useless, it is used in `utils.py`

File: flask_openapi3/scaffold.py (hunk 4 above)
Comment (luolingchun) [reply flask_openapi3/scaffold.py:52]: I think this code is more straightforward and easier to understand, and if you have a better way, I suggest reopening a PR to discuss.

Comment (CostcoFanboy) [reply flask_openapi3/scaffold.py:52]: Understood, I'm not sure if my way is technically better. I'll propose one and show what I mean later. Maybe you'll like it.

**Pull request closed without merging**